javav2/example_code/cloudwatch/src/main/java/com/example/cloudwatch/scenario/CloudWatchActions.java

```diff
@@ -0,0 +1,1057 @@
+                String customMetricName = rootNode.findValue("customMetricName").asText();
+
+                SingleMetricAnomalyDetector singleMetricAnomalyDetector = SingleMetricAnomalyDetector.builder()
+                    .metricName(customMetricName)
+                    .namespace(customMetricNamespace)
+                    .stat("Maximum")
+                    .build();
+
+                PutAnomalyDetectorRequest anomalyDetectorRequest = PutAnomalyDetectorRequest.builder()
+                    .singleMetricAnomalyDetector(singleMetricAnomalyDetector)
+                    .build();
+
+                return getAsyncClient().putAnomalyDetector(anomalyDetectorRequest).thenAccept(response -> {
+                    logger.info("Added anomaly detector for metric {}", customMetricName);
+                });
+            } catch (Exception e) {
+                throw new RuntimeException("Failed to create anomaly detector", e);
+            }
+        }).whenComplete((result, exception) -> {
+            if (exception != null) {
+                throw new RuntimeException("Error adding anomaly detector", exception);
+            }
+        });
+    }
+    // snippet-end:[cloudwatch.java2.scenario.add.anomalydetector.main]
+
+    // snippet-start:[cloudwatch.java2.scenario.get.alarm.history.main]
+
+    /**
+     * Retrieves the alarm history for a given alarm name and date range.
+     *
+     * @param fileName the path to the JSON file containing the alarm name
+     * @param date     the date to start the alarm history search (in the format "yyyy-MM-dd'T'HH:mm:ss'Z'")
+     * @return a {@code CompletableFuture<Void>} that completes when the alarm history has been retrieved and processed
+     */
+    public CompletableFuture<Void> getAlarmHistoryAsync(String fileName, String date) {
+        CompletableFuture<String> readFileFuture = CompletableFuture.supplyAsync(() -> {
+            try {
+                JsonParser parser = new JsonFactory().createParser(new File(fileName));
+                com.fasterxml.jackson.databind.JsonNode rootNode = new ObjectMapper().readTree(parser);
+                return rootNode.findValue("exampleAlarmName").asText(); // Return alarmName from the JSON file
+            } catch (IOException e) {
+                throw new RuntimeException("Failed to read or parse the file", e);
+            }
+        });
+
+        return readFileFuture.thenCompose(alarmName -> {
+            try {
+                Instant start = Instant.parse(date);
+                Instant endDate = Instant.now();
+                DescribeAlarmHistoryRequest historyRequest = DescribeAlarmHistoryRequest.builder()
+                    .startDate(start)
+                    .endDate(endDate)
+                    .alarmName(alarmName)
+                    .historyItemType(HistoryItemType.ACTION)
+                    .build();
+
+                return getAsyncClient().describeAlarmHistory(historyRequest).thenAccept(response -> {
```

Comment:
Where is the pagination?

Reply:
Updated the call to use describeAlarmHistoryPaginator()